vcstool/clients/vcs_base.py

```diff
@@ -93,7 +113,12 @@ def load_url(url, retry=2, retry_period=1, timeout=10):
     try:
         fh = urlopen(url, timeout=timeout)
     except HTTPError as e:
-        if e.code == 503 and retry:
+        if e.code in (401, 404):
+            # Try again, but with authentication
+            fh = _authenticated_urlopen(url, timeout=timeout)
+            if fh is not None:
+                return fh.read()
```

Comment:
Instead of duplicating the return statement how about changing the remaining of this block to only raise in the `else` case?

Reply:
Good idea. Refactored in 1cdc397997a4c9abd38717fb7cc08a4ada95d58a.